install/update_2.5_2.6.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ *
+ * @param Migration $migration
+ *
+ * @return void
+ */
+function plugin_formcreator_update_2_6(Migration $migration) {
+   global $DB;
+
+   $migration->displayMessage("Upgrade to schema version 2.6");
+
+   // update questions
+   $question = new PluginFormcreatorQuestion();
+   $rows = $question->find("`regex` <> '' AND `regex` IS NOT NULL");
+   $table = PluginFormcreatorQuestion::getTable();
+   foreach ($rows as $id => $row) {
+      // Avoid notice when validating the regex
+      set_error_handler(function($errno, $errstr, $errfile, $errline, $errcontext) {});
+      $isValid = !(preg_match($row['regex'], null) === false);
+      restore_error_handler();
+
+      if (!$isValid) {
+         // The regex is invalid (likely not updated yet)
+         $updatedRegex = addslashes('/^' . $row['regex'] . '$/');
```

Comment:
When an admin adds a regex validation,  the previous behavior is to detect if the regex has delimiters and anchors. If one of them was missing, then the plugin added them automatically before saving.

So, thee regexes in DB in versions up to 2.5.x always start with "/^" and ends "$/".

Actually, your comment makes me realize this upgrade code is useless. I need to remove it.